src/sqlfluff/core/rules/base.py

```diff
@@ -663,7 +663,7 @@ def crawl(
             new_lerrs: List[SQLLintError] = []
             new_fixes: List[LintFix] = []
 
-            if res is None:
+            if res is None or res == []:
```

Comment:
Bug fix: It should be legal for a rule to return an empty list of `ListResult`. Previously, this was crashing with an `IndexError` on line 681 below.